File: apply-migration.js
const { createClient } = require('@supabase/supabase-js');
require('dotenv').config();

const supabaseUrl = process.env.VITE_SUPABASE_URL;
const supabaseServiceKey = process.env.SUPABASE_SERVICE_ROLE_KEY || process.env.VITE_SUPABASE_ANON_KEY;

if (!supabaseUrl || !supabaseServiceKey) {
  console.error('❌ Erro: Credenciais do Supabase não configuradas');
  console.error('Verificar .env: VITE_SUPABASE_URL e SUPABASE_SERVICE_ROLE_KEY');
  process.exit(1);
}

async function applyMigration() {
  try {
    console.log('⏳ Conectando ao Supabase...');
    
    const supabase = createClient(supabaseUrl, supabaseServiceKey, {
      auth: { persistSession: false }
    });

    // Read the migration SQL
    const fs = require('fs');
    const migrationSQL = fs.readFileSync('supabase/migrations/20260623_split_stock_suppliers_address.sql', 'utf8');

    console.log('⏳ Aplicando migração...\n');

    // Execute using RPC or raw query
    const { error } = await supabase.rpc('exec', { 
      sql_string: migrationSQL 
    }).then(r => r, e => ({ error: e }));

    if (error && error.message && error.message.includes('exec')) {
      // RPC doesn't exist, try direct query approach
      console.log('ℹ️  RPC não disponível, tentando método alternativo...\n');
      
      // Split by semicolon and execute each statement
      const statements = migrationSQL.split(';').filter(s => s.trim());
      
      for (const statement of statements) {
        const trimmed = statement.trim();
        if (!trimmed) continue;
        
        const { error: execError } = await supabase.from('stock_suppliers').insert([{fake: true}]).select().then(
          () => ({ error: null }),
          () => ({ error: null }) // Just check connection
        );
      }
      
      console.log('⚠️  Não foi possível aplicar via API Supabase');
      console.log('Por favor execute manualmente no Supabase SQL Editor:\n');
      console.log(migrationSQL);
      process.exit(1);
    }

    if (error) {
      console.error('❌ Erro:', error.message);
      process.exit(1);
    }

    console.log('✅ Migração aplicada com sucesso!');
    console.log('\n📋 Próximos passos:');
    console.log('1. Atualize a página do navegador (F5)');
    console.log('2. Importe um fornecedor via XML do módulo de Contas a Pagar');
    console.log('3. Verifique se o campo "Rua" agora é preenchido automaticamente\n');

  } catch (err) {
    console.error('❌ Erro inesperado:', err.message);
    console.log('\nPor favor execute manualmente no Supabase SQL Editor:');
    const fs = require('fs');
    const migrationSQL = fs.readFileSync('supabase/migrations/20260623_split_stock_suppliers_address.sql', 'utf8');
    console.log(migrationSQL);
    process.exit(1);
  }
}

applyMigration();

File: apply_migration.py
import os
import psycopg2
from dotenv import load_dotenv
import sys

load_dotenv()

# Parse Supabase connection string from URL
supabase_url = os.getenv('VITE_SUPABASE_URL')  # https://xxxxx.supabase.co
supabase_key = os.getenv('VITE_SUPABASE_ANON_KEY')

if not supabase_url:
    print('❌ Erro: VITE_SUPABASE_URL não configurado')
    sys.exit(1)

# Extract host from URL: https://xxxxx.supabase.co -> xxxxx.supabase.co
host = supabase_url.replace('https://', '').replace('http://', '')

# Try to read the database password from environment
db_password = os.getenv('DB_PASSWORD')
if not db_password:
    print('⚠️  DB_PASSWORD não configurado')
    print('Por favor, execute manualmente o SQL no Supabase SQL Editor:')
    print()
    with open('supabase/migrations/20260623_split_stock_suppliers_address.sql', 'r') as f:
        print(f.read())
    sys.exit(1)

try:
    # Connect to Supabase PostgreSQL
    print('⏳ Conectando ao Supabase...')
    conn = psycopg2.connect(
        host=host,
        port=5432,
        database='postgres',
        user='postgres',
        password=db_password
    )
    
    cursor = conn.cursor()
    
    print('⏳ Aplicando migração para adicionar street e number campos...\n')
    
    # Read and execute migration SQL
    with open('supabase/migrations/20260623_split_stock_suppliers_address.sql', 'r') as f:
        migration_sql = f.read()
    
    cursor.execute(migration_sql)
    conn.commit()
    
    print('✅ Migração aplicada com sucesso!')
    print('\n📋 Próximos passos:')
    print('1. Atualize a página do navegador (F5)')
    print('2. Importe um fornecedor via XML do módulo de Contas a Pagar')
    print('3. Verifique se o campo "Rua" agora é preenchido automaticamente')
    
    cursor.close()
    conn.close()
    
except psycopg2.Error as e:
    print(f'❌ Erro ao conectar ao banco: {e}')
    print('\nPor favor, execute manualmente o SQL no Supabase SQL Editor:')
    with open('supabase/migrations/20260623_split_stock_suppliers_address.sql', 'r') as f:
        print(f.read())
    sys.exit(1)

except FileNotFoundError:
    print('❌ Arquivo de migração não encontrado')
    sys.exit(1)
